src/assessments/seo/keyphraseLengthAssessment.js

```diff
@@ -1,31 +1,99 @@
-var AssessmentResult = require( "../../values/AssessmentResult.js" );
-
+let Assessment = require( "../../assessment.js" );
+let AssessmentResult = require( "../../values/AssessmentResult.js" );
+const inRange = require( "lodash/inRange" );
+const merge = require( "lodash/merge" );
 /**
- * Assesses the keyphrase presence and length
- *
- * @param {Paper} paper The paper to use for the assessment.
- * @param {Researcher} researcher The researcher used for calling research.
- * @param {Jed} i18n The object used for translations
- * @returns {AssessmentResult} The result of this assessment
-*/
-function keyphraseAssessment( paper, researcher, i18n ) {
-	var keyphraseLength = researcher.getResearch( "keyphraseLength" );
-
-	var assessmentResult = new AssessmentResult();
-
-	if ( ! paper.hasKeyword() ) {
-		assessmentResult.setScore( -999 );
-		assessmentResult.setText( i18n.dgettext( "js-text-analysis", "No focus keyword was set for this page. " +
-			"If you do not set a focus keyword, no score can be calculated." ) );
-	} else if ( keyphraseLength > 10 ) {
-		assessmentResult.setScore( 0 );
-		assessmentResult.setText( i18n.dgettext( "js-text-analysis", "The keyphrase is over 10 words, a keyphrase should be shorter." ) );
+ * Assessment to check whether the keyphrase has a good length.
+ */
+class KeyphraseLengthAssessment extends Assessment {
+	/**
+	 * Sets the identifier and the config.
+	 *
+	 * @param {Object} config The configuration to use.
+	 *
+	 * @returns {void}
+	 */
+	constructor( config = {} ) {
+		super();
+
+		let defaultConfig = {
+			scores: {
+				keyphraseLength: 0,
+			},
+			assessment: {
+				recommendedMaximum: 5,
+				acceptableMaximum: 9,
+			},
+		};
+
+		this.identifier = "keyphraseLength";
+		this._config = merge( defaultConfig, config );
+	}
+
+	/**
+	 * Assesses the keyphrase presence and length.
+	 *
+	 * @param {Paper} paper The paper to use for the assessment.
+	 * @param {Researcher} researcher The researcher used for calling research.
+	 * @param {Jed} i18n The object used for translations.
+	 * @returns {AssessmentResult} The result of this assessment.
+	 */
+	getResult( paper, researcher, i18n ) {
+		this._keyphraseLength = researcher.getResearch( "keyphraseLength" );
+		let assessmentResult =  new AssessmentResult();
+
+		if ( paper.hasKeyword() ) {
+			assessmentResult.setScore( this.calculateScore() );
+			assessmentResult.setText( this.translateScore( i18n ) );
+		} else  {
+			assessmentResult.setScore( -999 );
+			assessmentResult.setText( i18n.dgettext( "js-text-analysis", "No focus keyword was set for this page. " +
+				"If you do not set a focus keyword, no score can be calculated." ) );
+		}
+		return assessmentResult;
+	}
+
+	/**
+	 * Calculates the score result based on the keyphraseLength research.
+	 * @returns {Object} object with score and text.
+	 */
+	calculateScore() {
+
+		if ( inRange( this._keyphraseLength, 1, this._config.assessment.recommendedMaximum ) ) {
```

Comment:
A problem with our current implementation of classes is that it forces you to repeat the same if statements for setting the score and setting the feedback message. A way to make this code a bit less redundant and more error-resistant is to write separate functions for the conditions of the if statements, see https://github.com/Yoast/YoastSEO.js/blob/a684b7bc129a27f2813591ca125aba3cc7b08475/src/assessments/seo/metaDescriptionKeywordAssessment.js